datadog_checks_base/datadog_checks/base/utils/db/utils.py

```diff
@@ -278,3 +278,71 @@ def _run_job_rate_limited(self):
 
     def run_job(self):
         raise NotImplementedError()
+
+
+def dbm_tracked_method(dbms, parent_check_attr=None, track_result_length=False):
+    """
+    Decorates a function to provide debug metrics and logging for troubleshooting.
+    Tracks execution time, errors, and result length.
+
+    If the check has a `debug_stats_kwargs` function, that function is called to get a set of kwargs to pass to
+    the statsd methods (i.e. histogram, count, gauge, etc). This is useful when specific tags need to be added to
+    these debug metrics in a standardized way.
+
+    :param dbms: the dbms to report as (i.e. sqlserver)
+    :param parent_check_attr: the attribute of the class which references the check. If none, self must refer to the
+    check. The check reference is used to report debug metrics and to generate logs.
+    :param track_result_length: if true, the length of the result is tracked
+    :return: a decorator
+    """
+
+    def decorator(function):
```

Comment:
If the intent is to have the tracking decorator enabled by default, there should be a config option (can be a hidden option) to disable the tracking on the user side

Reply:
good idea, added option to disable